### PR TITLE
[WIP] Feature: Add (Nested) Validation to Create / Update / Delete Args

### PR DESCRIPTION
**Pull request by johannesschobel**
Dear @doug-martin ,

this PR addresses the issue discussed in #19 .
in particular, the commits presented below add respective `@ValidateNested()` and `@Type()` decorators as discussed.

Please note, that this PR contains a **breaking change**, as i changed one `@Field` from `input` to `id` to better fit the overall API. This breaking change is, however, encapsulated in one single commit!

Further - I was **not able** to properly run all tests via `npm run test`. There are 2 tests related to `create one` and `create many` in the `create.resolver.spec` file that won't pass. If you can tell me, how to properly fix the tests (or adapt my implementation) i would be happy to re-work specific parts.

All the best and thank you very much for your time and effort!

### CODE REVIEW

**Comment by johannesschobel** [comment]
hmmmmmmmmmmmmmmmm... wait.. strange.. die travis CI build says it passes.. however, if you look into the output message, there are 2 tests failing!?
thats the same 2 tests that also fail on my machine

See here: https://travis-ci.org/doug-martin/nestjs-query/jobs/653485852#L394

File: packages/query-graphql/src/types/delete-one-args.type.ts [hunk 1]
```diff
@@ -13,8 +14,9 @@ export function DeleteOneArgsType(): Class<DeleteOneArgsType> {
   }
   @ArgsType()
   class DeleteOneArgs implements DeleteOneArgsType {
+    @IsNotEmpty()
     @Field(() => ID, { description: 'The id of the record to delete.' })
-    input!: string | number;
+    id!: string | number;
```
Comment (doug-martin) [inline packages/query-graphql/src/types/delete-one-args.type.ts]: You should update the docs to reflect this change.

**Comment by doug-martin** [comment]
Hmm not sure why travis is showing as passing, Ill have to take a look at that. I think you can change the deepEqual to a objectContaining from `ts-mockito` and it might fix it. I think the difference is that it is actually transforming the input into the correct type now.

**Comment by doug-martin** [comment]
I think the failing test issue should be resolved now. If you can rebase and push back up we can test it out.

**Comment by doug-martin** [comment]
@johannesschobel Im going to issue a new PR with your changes merged and fixed tests.

One thing Im going to revert is the renaming input to `id`. Ill do this in another PR but Im going to follow the recommendations outlined in https://blog.apollographql.com/designing-graphql-mutations-e09de826ed97

**Comment by doug-martin** [comment]
Sorry I'd didnt explain why...I need the changes also for a project I'm working on also.

**Comment by johannesschobel** [comment]
sure, no problem.. glad i could help (although you did most of it, haha)

**Pull request closed without merging**